src/core_algorithms.jl

```diff
@@ -69,17 +69,28 @@ end
     elseif length(sites) == 1 # Same if there is a single site
         view(grid, TL[1]:BR[1], TL[2]:BR[2]) .= Ref(first(sites))
     else
-        # Otherwise we check if all corners have the same closest site
+        # Otherwise we check if all corners have the same closest site ...
```

Comment:
This part I don't understand

Reply:
If you modify the code here like

```
            if all(zip(mins, dists)) do ((min_dist, _), dist)
                    dist > min_dist
                end
                grid[TL[1]:BR[1], TL[2]:BR[2]] .= Ref(min_site)
                return nothing
            else
                @show (distance, mins, dists)
                # grid[TL[1]:BR[1], TL[2]:BR[2]] .= Ref(min_site)
                return nothing
            end
```
you'll see the non unique sites and if you also set the rectangle to the the site you'll see the test failures for Chebyshev distance. BTW we could also drop the whole code sequence, because filtering is good enough (see the benchmark for different find and filter combinations in legacy).